packages/client/src/containers/CollectionsPage/CollectionsPage.Container.js

```diff
@@ -54,31 +54,29 @@ export const CollectionsPage = () => {
   const sortOptions = [
     'Recent Collections',
     'Alphabetically',
-    'Price ↓',
-    'Price ↑',
+    'Price Low to High',
+    'Price High to Low',
   ];
 
-  useEffect(() => {
-    function sortFunction(a, b) {
-      if (sortOrder === '') {
-        return 0;
-      }
-      if (sortOrder === 'Recent Collections') {
-        return a.created_at > b.created_at ? -1 : 1;
-      }
-      if (sortOrder === 'Alphabetically') {
-        return a.name.toLowerCase() < b.name.toLowerCase() ? -1 : 1;
-      }
-      if (sortOrder === 'Price ↓') {
-        return Number(a.price) > Number(b.price) ? 1 : -1;
-      }
-      if (sortOrder === 'Price ↑') {
-        return Number(a.price) < Number(b.price) ? 1 : -1;
-      }
+  function sortFunction(a, b) {
+    if (sortOrder === '') {
       return 0;
     }
-    filteredProducts.sort(sortFunction);
-  }, [filteredProducts, sortOrder]);
+    if (sortOrder === 'Recent Collections') {
+      return a.created_at > b.created_at ? -1 : 1;
+    }
+    if (sortOrder === 'Alphabetically') {
+      return a.name.toLowerCase() < b.name.toLowerCase() ? -1 : 1;
+    }
+    if (sortOrder === 'Price Low to High') {
+      return Number(a.price) > Number(b.price) ? 1 : -1;
+    }
+    if (sortOrder === 'Price High to Low') {
+      return Number(a.price) < Number(b.price) ? 1 : -1;
+    }
+    return 0;
+  }
+  filteredProducts.sort(sortFunction);
```

Comment:
I'm not sure if this is the right thing to do though. 

This way, the list is sorted on **every** render even if the list or the sort order didn't change. The usage of `useEffect` prevents that, why did you feel it was needed to remove it?

Reply:
Thank you for feedback. I have tried all way. Its come same result with bug when I used useEffect and I think its come error because after fetching data from API, Its store in products array and products array filter with search input and store with filteredProducts array. filteredProduct store Valued, I used for sorting in function.Thats reason useEffect is not working. If I use useEffect with products Array which is fetch data from API than Its working perfectly than I get problem with search bar Its not working.  I am working on one another solution, I am trying to make sortOrder endpoint where I can sort producs by fetch API.

Reply:
Yaah, You are right, I am not write commit comment properly. Now I have cleared for how to write sort message.

Reply:
If you're having issue with many parts of state conflicting that usually means you have to lift more of the state higher up. So that the query string, the products, the sort order, the API call, happen in the same place so that the `useEffect` can react properly. 
Moving the sorting to the API is one solution but I would try first to learn how to keep your state in sync since you'll face similar challenges in the future.